gae_serve/README.md

```diff
@@ -0,0 +1,134 @@
+
+1. Deploy the service endpoint:
+
+    `gcloud service-management deploy modelserve.yaml`
+
+    This step deploys a [Cloud Endpoint](https://cloud.google.com/endpoints/) service, which allows us to monitor the API usage on the [Endpoints](https://console.cloud.google.com/endpoints) console page.
+
+1. If the deployment is successful, get the deployment's config id either from the [Endpoints](https://console.cloud.google.com/endpoints) console page under the service's Deployment history tab, or you can find all the configuration IDs by running the following:
+
+    `gcloud service-management configs list --service="PROJECT_ID.appspot.com"`
+
+    The configuration ID should look like `2017-08-03r0`.  The `r0, r1, ...` part in the configuration IDs indicate the revision numbers, and you should use the highest (most recent) revision number.
+
+1. Create a GCS bucket with your choice of a `BUCKET_NAME`, and copy the sample model file over:
+
+    ```
+    gsutil mb gs://BUCKET_NAME
+    gsutil cp lr.pkl gs://BUCKET_NAME
+    ```
+
+1. Update `app.yaml`:
+
+    * If you already have at least one GAE service in your GCP project:
+
+        - Replace `PROJECT_ID` with your GCP project's id.
+
+        - Replace `BUCKET_NAME` with the name of the bucket you created on GCS above.
+
+        - Replace `CONFIG_ID` with the configuration ID you got from the service endpoint deployment.
+
+    * If you have not deployed any service to GAE with this GCP project:
+
+        - Replace `service: modelserve` with `service: default` in `app.yaml`.  The first service deployed to GAE must be named `default`.
+
+        - Replace `PROJECT_ID` with your GCP project's id.
+
+        - Replace `BUCKET_NAME` with the name of the bucket you created on GCS above.
+
+        - Replace `CONFIG_ID` with the configuration ID you got from the service endpoint deployment.
+
+        - **Note that you will not be able to delete the `default` service from your project.**
+
+    See the [documentation](https://cloud.google.com/appengine/docs/flexible/python/configuring-your-app-with-app-yaml) for more information about `app.yaml`.
+
+1. Deploy the backend service:
+
+    `gcloud app deploy`
+
+    **This step could take several minutes to complete.**
+
+
+1. If the deployment is successful, you can test it from the command line:
+
+    - Create an API key with the "Create credentials" button on the [Credentials](https://console.cloud.google.com/apis/credentials) page.  Make sure you switch to the correct GCP project first.
+
+    - Now you can test the service endpoint: (Remember to replace `PROJECT_ID` and `API_KEY` with their actual values below.)
+
+        * If you have `service: modelserve` in `app.yaml`:
+
+            `curl -H "Content-Type: application/json" -X POST -d '{"X": [[1, 2], [5, -1], [1, 0]]}' "https://modelserve-dot-PROJECT_ID.appspot.com/predict?key=API_KEY"`
```

Comment:
Could we actually include a little Python client to make this request? Not sure how well this would work for Windows users.

Reply:
This would in general also be useful as it would give them a template client that they could use to make requests against their service - teaching, for example, how to read `API_KEY` (from environment? from config?) and similarly for `PROJECT_ID`.

Reply:
Might be good to ask @SurferJeffAtGoogle to go through this whole workflow once and measure the amount of rage induced.

Reply:
Adding a client is a great idea - using swagger/OpenAI should make it easy.  I will do this before merge.